recipes/libzip/all/conanfile.py

```diff
@@ -0,0 +1,111 @@
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_bzip2": [True, False],
+        "with_lzma": [True, False],
+        "with_zstd": [True, False],
+        "crypto": [False, "win32", "openssl", "mbedtls", "auto"]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_bzip2": True,
+        "with_lzma": True,
+        "with_zstd": True,
+        "crypto": "auto"
+    }
+
+    @property
+    def _crypto(self):
+        if self.options.crypto == "auto":
+            return "win32" if self.settings.os == "Windows" else "openssl"
+        return self.options.crypto
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self._crypto == "win32" and self.settings.os != "Windows":
+            raise ConanInvalidConfiguration("Windows is required to use win32 crypto libraries")
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        self.requires("zlib/1.2.11")
+
+        if self.options.with_bzip2:
+            self.requires("bzip2/1.0.8")
+
+        if self.options.with_lzma:
+            self.requires("xz_utils/5.2.4")
+
+        if self.options.with_zstd:
+            self.requires("zstd/1.4.5 ")
+
+        if self._crypto == "openssl":
+            self.requires("openssl/1.1.1g")
+        elif self._crypto == "mbedtls":
+            self.requires("mbedtls/2.16.3-gpl")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["BUILD_REGRESS"] = False
+        cmake.definitions["BUILD_EXAMPLES"] = False
+        cmake.definitions["BUILD_DOC"] = False
+
+        cmake.definitions["ENABLE_LZMA"] = self.options.with_lzma
+        cmake.definitions["ENABLE_BZIP2"] = self.options.with_bzip2
+        cmake.definitions["ENABLE_ZSTD"] = self.options.with_zstd
+
+        cmake.definitions["ENABLE_COMMONCRYPTO"] = False  # TODO: We need CommonCrypto package
+        cmake.definitions["ENABLE_GNUTLS"] = False  # TODO: We need GnuTLS package
+
+        cmake.definitions["ENABLE_MBEDTLS"] = self._crypto == "mbedtls"
+        cmake.definitions["ENABLE_OPENSSL"] = self._crypto == "openssl"
+        cmake.definitions["ENABLE_WINDOWS_CRYPTO"] = self._crypto == "win32"
+
+        cmake.configure()
+        self._cmake = cmake
+
+    def build(self):
+        self._configure_cmake()
+        self._cmake.build()
+
+    def package(self):
+        self._cmake.install()
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
```

Comment:
You're removing cmake and pkg_config files here.
Can you explicitly set the names of the cmake_find_config(_multi) and pkg_config in package_info?

Reply:
Should I still delete them from package?

Reply:
I think I have done it. (copy-pasted from other recipes) 
Is this what you meant: https://github.com/conan-io/conan-center-index/pull/2804/files#diff-0573a88da3ee27dee80eb51dd9b53683R106-R108 ? Thank you.

Reply:
Yes, that's what I meant.

cci want to generate these files itself because these might contain paths to the build machine.
It is possible to generate relocatable cmake find package modules, so for them it is not always neded.
But the pc files often contain a root path.